core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3348,14 +3348,25 @@ class ViewMapping<
   //----------------------------------------
 
   KOKKOS_DEFAULTED_FUNCTION ~ViewMapping() = default;
-  KOKKOS_INLINE_FUNCTION ViewMapping() : m_impl_handle(), m_impl_offset() {}
+  KOKKOS_DEFAULTED_FUNCTION ViewMapping()  = default;
```

Comment:
Why did you change the default constructor (when needing to provide initialization defaults anyway)?

Reply:
Because I find it clearer to have proper default member values alongside with the member definition (see your comment above).

Also, in #6524, @dalg24 didn't use `KOKKOS_DEFAULTED_FUNCTION` for the default `Bitset` constructor. And in fact `KOKKOS_DEFAULTED_FUNCTION` is always defined empty. So is there any reason why `Kokkos` is keeping this macro alive? Should it be deprecated?

Reply:
I did not use `KOKKOS_DEFAULTED_FUNCTION` because `Bitset` is not intended to be default constructible on the device.

Reply:
> I did not use KOKKOS_DEFAULTED_FUNCTION because Bitset is not intended to be default constructible on the device.

Ah, OK! It's clearer now :mag: